### ISSUE TITLE
Multipart/form-data : deserialize issue

### ISSUE THREAD

**Issue by jy95** [byu-oit/openapi-enforcer-middleware#21]
Hello,

First, thank you @Gi60s for this cleary good documented module.
I have a issue with this module for an [endpoint](https://github.com/SourceCodeOER/sourcecode_api/blob/temp-master/openapi/paths/bulk.yaml#L115) that consumes either `application/json` or `multipart/form-data` ( depending of the context ).

I fixed the bug of your multer middelware in with my [PR](https://github.com/byu-oit/openapi-enforcer-multer/pull/4) but I found another bug in `openapi-enforcer-middleware` that is hard to detect ^^.

After `openapi-enforcer-multer` does its job ,my `req.body` looks like this : 
![image](https://user-images.githubusercontent.com/9306961/71322848-a47ff900-24cc-11ea-8d07-626b5946565f.png)

However when applying `enforcer.middleware()`, I didn't find how but the deserialize failed ...
![image](https://user-images.githubusercontent.com/9306961/71322946-be6e0b80-24cd-11ea-9d05-f61682471f0c.png)

![image](https://user-images.githubusercontent.com/9306961/71322985-7dc2c200-24ce-11ea-9edc-fc2d86c3cacf.png)

If I replace 
```js
const name = (value.constructor).name;
```
by
```js
const name = (value.constructor || {}.constructor).name;
```

It shows the failed validation : 

![image](https://user-images.githubusercontent.com/9306961/71323041-253ff480-24cf-11ea-8c31-d65b285c8325.png)

It seems strange as the data ( except the integer field in "filesMapping" isn't converted to an integer ) seems to respect request body schema ...

**Comment by jy95** [comment]
I also checked the state of the code before invocation of `const [ request, clientError ] = openapi.request(requestObj, { allowOtherQueryParameters: this.options.allowOtherQueryParameters })` : 

![image](https://user-images.githubusercontent.com/9306961/71323207-de52fe80-24d0-11ea-9779-d0504280e8a0.png)

So I cannot understand why there is this serialize/deserialize issue ....

**Comment by Gi60s** [comment]
Would it be possible to modify your code (or comment out lines) so that it doesn't use SQL. I'm trying to troubleshoot but I'm having a hard time getting my environment set up to work with the temp-master branch.

Alternatively maybe you can set up a docker compose file that sets up the environment.

**Comment by jy95** [comment]
@Gi60s An better way to reproduce the bug is to take only the required things so you can reproduce that without all the embedded sql (as I work on it, I don't necessarily think I'll touch it.).

A very summarized OpenAPI specification of my problem can be  : 

**api.yaml ( the endpoint ) :** 

```yaml
openapi: 3.0.0
info:
  description: "API for Source Code"
  version: "1.0.0"
  title: "Source Code"
  license:
    name: "GPL-3.0-or-later"
    url: "https://choosealicense.com/licenses/gpl-3.0/"

# Pas utile pour l'instant mais c'est mieux pour après
servers:
  - url: http://localhost:3000
    description: Internal staging server for testing
  - url: http://api.example.com/v1
    description: Optional server description, e.g. Main (production) server

paths:
  /api/bulk/create_exercises:
    $ref: 'openapi/paths/bulk.yaml#/paths/~1api~1bulk~1create_exercises'
```

**The path file ( it normally contained more of that but I simplified it ) :** 

```yaml
paths:
  /api/bulk/create_exercises:
    post:
      summary: "Creates multiple exercises into the system"
      operationId: createMultipleExercises
      x-controller: bulk
      x-operation: createMultipleExercises
      security:
        - bearerAuth: []
      requestBody:
        required: true
        content:
          application/json:
            schema:
              $ref: "../definitions.yaml#/components/schemas/MultipleExercisesForm"
          multipart/form-data:
            schema:
              type: object
              properties:
                exercisesData:
                  $ref: "../definitions.yaml#/components/schemas/MultipleExercisesForm"
                files:
                  type: array
                  description: "The sources of the exercises, each in zip format"
                  minItems: 1
                  items:
                    type: string
                    format: binary
                    description: "The source of an exercise, in zip format"
                filesMapping:
                  type: array
                  description: "Mapping between the given file (to find out which data belong to exercise)"
                  minItems: 1
                  items:
                    type: object
                    properties:
                      filename:
                        type: string
                        description: "The name of the given source file"
                        example: "file1.zip"
                      exercise:
                        type: integer
                        description: "The location of the related exercise in the exercisesData array"
                    required:
                      - filename
                      - exercise
              required:
                - exercisesData
                - files
                - filesMapping
      responses:
        '200':
          description: OK
        # Definition of all error statuses
        default:
          description: "Whatever error : 4XX - Client error (Bad Request, Unauthorized, etc.) , 5XX - Server error"
          content:
            application/json:
              schema:
                $ref: "../definitions.yaml#/components/schemas/ErrorObject"

```

**definitions.yaml (all my commonly defined things) **

```yaml
components:
  schemas:
    ErrorObject:
      type: object
      properties:
        message:
          type: string
          description: The main error message ( for example "Bad Request", "Unauthorized", etc. )
        errors:
          type: array
          items:
            type: object
            description: Explanation about an error
      required:
        - message
        - errors
# Basic model of a Exercise 
    BasicExerciseModel:
      type: object
      properties:
        title:
          type: string
          example: "A Super Exercise"
          minLength: 3
          maxLength: 100
          description: "The title of this exercise"
        description:
          type: string
          maxLength: 5000
          example: "..."
          description: "The preamble of this exercise"
      required:
        - title
        - description
    MultipleExercisesForm:
      type: array
      minItems: 1
      items:
        $ref: "#/components/schemas/ExerciseForm" 
        description: "An array of exercises"
# When we want to upload a new exercise, we need other fields for that
    ExerciseForm:
      allOf:
        - $ref: "#/components/schemas/BasicExerciseModel"
        - type: object
          properties:
            tags:
              type: array
              items:
                oneOf:
                  - type: integer
                    minimum: 0
                    description: "A Tag ID ( already existent in database )"
                  - $ref: "#/components/schemas/TagProposal"
                    description: "A not-existent Tag we want to add"
              description: "Mixed array that contains existent tag(s) or not"
              uniqueItems: true
              minItems: 1 # We must always put at least one tag in the database
            url:
              type: string
              pattern: '^https?:\/\/[^\s$.?#].[^\s]*$'
              example: "https://inginious.info.ucl.ac.be/course/LEPL1402/Streams"
              nullable: true
              description: "If not null, the link to the exercise on a plateform"
          required:
            - tags
```

**Comment by jy95** [comment]
@Gi60s With this small open api documentation and this request example, I think it should be enough to reproduce the bug ( if it passes the `openapi-enforcer-middleware` with the expected result, it should work as expected ). 
Feel free to ask if you encounter any other difficulty

**Comment by jy95** [comment]
@Gi60s I created a [minimal demo repository](https://github.com/jy95/demoBug) to better showcase that issue. Much more simpler for you to test ;)

Feel free to ask if you encounter any other difficulty

**Comment by jy95** [comment]
@Gi60s I found a [middleware workaround](https://github.com/jy95/demoBug/blob/master/app.js#L50) : you can see [it works](https://travis-ci.com/jy95/demoBug/builds/142297205#L301) but if there is a cleaner way, 
feel free to find one ;)

Edit : It seems that I found another issue in the `openapi-enforcer-multer` : It should be more simpler that this strange deserialization issue XD

**Comment by Gi60s** [comment]
I was able to fix this. The real problem was being obscured by an unexpected error in the error report. Once that was completed I realized that the [openapi-enforcer-multer](https://github.com/byu-oit/openapi-enforcer-multer) was not processing the non file `form-data` fields. To process those I also had to make an update to the [openapi-enforcer](https://github.com/byu-oit/openapi-enforcer) to enable a less strict deserialization. Processing the openapi-enforcer-multer `form-data` fields using less strict deserialization fixes the issue.

Update your dependencies in your minimal demo, remove the work around, and run the tests. It should all work.

 able to successfully deserialize the other `form-data` fields because
